File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp={}
        def slove(n):
            if n==1:
                return 1
            if n==2:
                return 2
            if n in dp:
                return dp[n]
            dp[n]=slove(n-1)+slove(n-2)
            return dp[n]
        return slove(n)

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:

    int ans(int n,vector<int> &dp)
    {
        if(n==0 || n==1)
        return 1;

        else if(n==2)
        return 2;

        if(dp[n]!=-1)
        return dp[n];

        dp[n]=ans(n-1,dp)+ans(n-2,dp);
        return dp[n];   
    }     
    int climbStairs(int n) {
        vector<int> dp(n+1,-1);
        return ans(n,dp);
    }
};